sftp-app-starters-source-common/src/main/java/org/springframework/cloud/stream/app/sftp/common/source/SftpSourceRotator.java

```diff
@@ -86,6 +92,15 @@ public String getCurrentDirectory() {
 		}
 		this.rotationPolicy.afterReceive(result != null, source);
 		return result;
+	}
 
+	/**
+	 * Evaluate the headers.
+	 * @return a {@code Map<String,Object>}
+	 */
+	public Map<String, Object> evaluateHeaders() {
```

Comment:
There is an `ExpressionEvalMap` in Spring Integration for exactly this purpose.
See its JavaDocs and usage in the code.